src/server/game/Movement/PathGenerator.cpp

```diff
@@ -942,3 +942,10 @@ void PathGenerator::ReducePathLenghtByDist(float dist)
         nextVec = currVec; // we're going backwards
     }
 }
+
+bool PathGenerator::IsInvalidDestinationZ(Unit* target)
+{
+    G3D::Vector3 endPos = GetActualEndPosition();
+
+    return (target->GetPositionZ() - endPos.z) > 5.f;
```

Comment:
replace both lines with `(target->GetPositionZ() - GetActualEndPosition().z) > 5.0f;` and add proper const to function (don't forget adjusting declaration in PathGenerator.h)
`bool PathGenerator::IsInvalidDestinationZ(Unit const* target) const`

for final effect
```
bool PathGenerator::IsInvalidDestinationZ(Unit const* target) const
{
    (target->GetPositionZ() - GetActualEndPosition().z) > 5.0f;
}
```